youtube_dl/extractor/svt.py

```diff
@@ -214,26 +215,44 @@ def _real_extract(self, url):
                 group='json'),
             video_id, fatal=False)
 
-        thumbnail = self._og_search_thumbnail(webpage)
+        info_dict = {}
 
         if data:
             video_info = try_get(
                 data, lambda x: x['context']['dispatcher']['stores']['VideoTitlePageStore']['data']['video'],
                 dict)
             if video_info:
                 info_dict = self._extract_video(video_info, video_id)
+                info_dict['title'] = data['context']['dispatcher']['stores']['MetaStore']['title']
+                self._adjust_title(info_dict)
+
+        if not info_dict:
+            svt_id = self._search_regex(
+                r'<video[^>]+data-video-id=["\']([\da-zA-Z-]+)',
+                webpage, 'video id')
+
+            info_dict = self._extract_by_video_id(svt_id, webpage)
+
+        if data:
+            video_data = try_get(data, lambda x: x['videoPage']['video'], dict)
+            if video_data:
+                if video_data.get('episodic'):
+                    info_dict.update({
+                        'season_number': video_data.get('season'),
+                        'episode_number': video_data.get('episodeNumber'),
+                    })
```

Comment:
1. `info_dict` may already contain correct values that will be overwritten by these possibly invalid ones.
2. Other metadata should also be extracted from this source.
3. `merge_dicts` for safe merging with priority.